manager/controllers/app/plotter_controller_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+package app
+
+import (
+	"context"
+	"time"
+
+	app "github.com/mesh-for-data/mesh-for-data/manager/apis/app/v1alpha1"
+	"github.com/mesh-for-data/mesh-for-data/manager/controllers/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Deploy the plotter and check that the result is what was expected.
+func deployAndCheck(namespace string, shouldSucceed bool) {
+	const timeout = time.Second * 30
+	const interval = time.Millisecond * 100
+
+	// Create the plotter yaml from the hard coded testdata file
+	plotter := &app.Plotter{}
+	Expect(readObjectFromFile("../../testdata/plotter.yaml", plotter)).ToNot(HaveOccurred())
+
+	// Set the namespace we received
+	plotter.SetNamespace(namespace)
+
+	// Create the Plotter
+	Expect(k8sClient.Create(context.Background(), plotter)).Should(Succeed())
```

Comment:
Should each  one be a test on itself? otherwise success and failures will not be deterministic.
I usually one test to check on condition, otherwise, it would be hard to connect a problem with a failure.

Reply:
Split into two separate tests